core/src/runtime/mod.rs

```diff
@@ -92,22 +94,59 @@ pub struct Runtime {
     pub should_report: bool,
 }
 
+/// Integer type used to count instructions and syscalls for `ExecutionReport`.
+type Count = u64;
+
 #[derive(Default, Debug, Clone, PartialEq, Eq)]
 pub struct ExecutionReport {
-    pub instruction_counts: HashMap<Opcode, u64>,
-    pub syscall_counts: HashMap<SyscallCode, u64>,
+    pub instruction_counts: HashMap<Opcode, Count>,
+    pub syscall_counts: HashMap<SyscallCode, Count>,
 }
 
 impl ExecutionReport {
-    pub fn total_instruction_count(&self) -> u64 {
+    pub fn total_instruction_count(&self) -> Count {
         self.instruction_counts.values().sum()
     }
 
-    pub fn total_syscall_count(&self) -> u64 {
+    pub fn total_syscall_count(&self) -> Count {
         self.syscall_counts.values().sum()
     }
 }
 
+/// Helper function. "Derives" `+=` if the value type of a `HashMap` supports it.
```

Comment:
nit: unecessary to say helper function just say

`/// Sums two hashmaps that keep track of counters together.`